.github/workflows/ci.yaml

```diff
@@ -8,8 +9,42 @@ on:
 env:
   CARGO_TERM_COLOR: always
 
+# Region tags below will fold blocks into functional areas
+# in vscode, # click to the left of the region tag to close the region.
+# more info: https://code.visualstudio.com/docs/editing/codebasics#_folding
+#
+# in vim, Use zR to open everything, zo=open one, zc=close one
+# For more information, type :help folding 
+
+#region Build Matrix
+# The build/test matrix is as follows:
+#
+#            ----------------------Profile & features-------------------------------- 
+#           +---------------------+------------------------+-------------------------+
+#           | debug & no features | debug & ethhash+logger | maxperfb ethhash+logger |
+#           +---------------------+------------------------+-------------------------+
+# CacheKey  | debug-no-features   | debug-ethhash          | maxperf-ethhash         |
+# license   | Yes (A)             | No                     | No                      |
```

Comment:
Would it make sense to use a matrix to specify the set of features + release profile throughout?

We can specify arbitrary variables for our own use, so that we don't need to replicate jobs with these two attributes as the only difference

Reply:
I tried that but it was actually less readable. Each job is subtly
different, whether it be the cache key or the command options being run.

Another thought was to generate this ci file from some code, maybe rust or
something, but that also seemed more complicated.